src/capi/azext_capi/custom.py

```diff
@@ -129,25 +130,34 @@ def find_kubectl_current_context():
         output = run_shell_command(command)
         output = output.strip()
         return output
-    except subprocess.CalledProcessError:
-        return None
+    except subprocess.CalledProcessError as err:
+        if "current-context is not set" in err.stdout:
+            return None
+        raise
 
 
-def find_cluster_in_current_context(context_name):
+def find_attribute_in_context(context_name, attribute="cluster"):
     command = ["kubectl", "config", "get-contexts", context_name, "--no-headers"]
+    result = None
     try:
         output = run_shell_command(command)
-        cluster_name = output.split()[2]
-        return cluster_name
+        output = output.split()
+        if attribute == "cluster":
+            result = output[2]
+        elif attribute == "user":
+            result = output[3]
+        elif attribute == "namespace":
+            result = output[4]
     except subprocess.CalledProcessError:
-        return None
+        result = None
```

Comment:
If we got here, then `result` should still equal `None`, so it's not strictly necessary to reassign it. I might replace this line with just `pass`.